### PR TITLE
On target tests - basic fixes

### PR DESCRIPTION
**Pull request by AndrzejKurek**
This PR adds necessary (but not all) changes needed to run the test suites on-target. The rest of changes will be added to the test and project repository.
It is easier to review them commit by commit.

### CODE REVIEW

File: tests/Makefile [hunk 1]
```diff
@@ -152,20 +152,25 @@ check: $(BINARIES)
 
 test: check
 
+prepare:
+	echo "  Prep  ../TESTS"
+	mkdir -p "../TESTS/host_tests"
```
Comment (shelib01) [inline tests/Makefile:157]: @AndrzejKurek , how and where this TEST folder was created previously ?

File: tests/suites/helpers.function [hunk 2]
```diff
@@ -601,7 +601,7 @@ static int uecc_rng_wrapper( uint8_t *dest, unsigned int size )
 }
 #endif /* MBEDTLS_USE_TINYCRYPT */
 
-#if defined(MBEDTLS_ENTROPY_HARDWARE_ALT)
+#if defined(MBEDTLS_ENTROPY_HARDWARE_ALT) && !defined(DEVICE_TRNG)
```
Comment (shelib01) [inline tests/suites/helpers.function:604]: Will we run we device TRNG if it is defined ?

**Review by shelib01 — APPROVED** [review]

File: tests/Makefile (hunk 1 above)
Comment (AndrzejKurek) [reply tests/Makefile:157]: I would assume that it was created in python test generating scripts (and we might not have to create the folder ourselves if we run them first), but it is better to have it explicitly created to note that we need additional preparation - like the script that gets copied.

File: tests/suites/helpers.function (hunk 2 above)
Comment (AndrzejKurek) [reply tests/suites/helpers.function:604]: If it is defined then the hardware poll function is defined here: https://github.com/PelionIoT/pmi-arm-firmware/blob/master/libs/mbed-os/features/mbedtls/platform/src/mbed_trng.cpp

**Comment by AndrzejKurek** [comment]
It seems that the merge job failed because of broken connection. Would you like to rerun it, @shelib01?

**Pull request merged**